controller/tests/controller_server/utils_test.py

```diff
@@ -28,10 +27,33 @@ def setUp(self):
                                       "publish_context_array_iqn": "array_iqn",
                                       "publish_context_fc_initiators": "fc_wwns"}
                        }
+        self.util_method = ""
+
+    def _test_validation_exception(self, method_arg, msg="", raised_error=ValidationException):
+        with self.assertRaises(raised_error) as ex:
+            getattr(utils, self.util_method)(method_arg)
+        if msg:
+            self.assertIn(msg, str(ex.exception))
+
+    def _test_validate_node_id_validation_exception(self, node_id):
+        self.util_method = "_validate_node_id"
```

Comment:
I don't think we should test private functions.
and I think it would be simpler to pass the function itself, instead of its name

Reply:
its just like validate_secret, I just dont understand why all the specific validation functions are private..

Reply:
and done with the "function itself"

Reply:
they are private because they are not called directly from outside the file

Reply:
then why is `self.util_method` [still initialized as a string](https://github.com/IBM/ibm-block-csi-driver/blob/beb4cc25b7fd99c888e8ba2656ebf28d71facd6f/controller/tests/controller_server/utils_test.py#L30)?
I don't think we need it in `self`

Reply:
I dont want to send it for every call..

Reply:
you set it up before every call, so it doesn't really add code.
and it's hard to see which method is being tested for each call (especially when it is set before other "preparations")

Reply:
it would also eliminate the need for `_test_validate_node_id_validation_exception`

Reply:
done

Reply:
the `util_function` is still being [set up before the call](https://github.com/IBM/ibm-block-csi-driver/blob/26f5c8277053a94c8d9ba26e7fb46a562302cc9b/controller/tests/controller_server/utils_test.py#L150), so it's still hard to see which method is being tested for each call (the argument name should be more specific than the generic parameter name).

if you don't want to function names in one line (e.g. `self._test_validation_exception(utils.validate_csi_volume_capabilities, [])`), then consider wrapping it (e.g. `self._test_validate_csi_volume_capabilities_validation_exception([])`). same for all `self._test_validation_exception` calls

Reply:
I [wrapped it](https://github.com/IBM/ibm-block-csi-driver/pull/425#discussion_r773858554), you asked to remove it..

Reply:
done

Reply:
I didn't ask to remove, I meant that two function names in one line could eliminate the wrapping need, then I saw a "done" without two function names in one line, so back to wrapping